### ISSUE TITLE
[stable/jenkins] Unable to mount volumes for pod after using fsgroup

### ISSUE THREAD

**Issue by mmiranda** [helm/charts#22857]
**Describe the bug**
Timeout when mounting volumes due to fsgroup

**Version of Helm and Kubernetes**: 
EKS 1.15

**Which chart**: stable/jenkins 

**What happened**:
After migrating to version >= 2 when starting to use `fsgroup`, pods gets timeout when trying to mount my `pvc` possibly due to the volume size (~70GB)

```Unable to mount volumes for pod "jenkins-8558c9669-nbmkr_jenkins(55aa3121-efff-444d-9db2-95483c4a2569)": timeout expired waiting for volumes to attach or mount for pod "jenkins"/"jenkins-8558c9669-nbmkr". list of unmounted volumes=[jenkins-home]. list of unattached volumes=[plugins tmp jenkins-config plugin-dir jenkins-home sc-config-volume jenkins-token-4zsr9]; skipping pod```

**What you expected to happen**:
Pods launching correctly like before bumping version

**How to reproduce it** (as minimally and precisely as possible):
Have a PVC to store jenkins data with a considerable size and  try to use fsgroup

**Anything else we need to know**:
as a Workaround I had to unset fsgroup
https://support.cloudbees.com/hc/en-us/articles/360035837431-Timeout-when-attaching-volumes-in-Kubernetes

**Comment by delphe** [comment]
I had the same issue. My workaround was using an older version by tacking on `--version "1.27.0"` during the helm install.

**Comment by mmiranda** [comment]
@delphe you still can use the latest helm version, just make sure to set `fsgroup=0` and `runAsUser=0` on your values

**Comment by fl-max** [comment]
There is a related Kubernetes issue [here](https://github.com/kubernetes/kubernetes/issues/67014). The **TL;DR** is that it's running a `chown` on all the files when `fsGroup` is set to something other than `0` which is a **very** lengthy process when you have millions(?) of files.

Running containers as root is not advisable. We've run into this same issue (not intrinsic to Jenkins) and just waited it out. For us it takes ~1.5hrs and eventually the Pod comes up.

As a mitigation strategy, we plan to do aggressive job clean up and cache deletion.

**Comment by graillus** [comment]
Just had the same issue when upgrading chart to 2.x, using the new `fsGroup=1000` config.

I followed the upgrade instructions from the README, the `chown` command took a huge amount of time compared to what I'm used to for a similar amount of files.

Then I applied the `helm upgrade`, and the Jenkins pod failed due to the mount timeout. I initially thought something was broken but then I saw this Github issue and waited, it took half an hour but finally worked without further intervention.

I'm running Jenkins on a GKE cluster, with standard persistent disk storage. I guess the relatively low disk I/O performance is the direct cause of the slowness
